Regular-Question-Answers/merge-sorted-array.cpp

```diff
@@ -0,0 +1,35 @@
+// Author: Ali Emre Kaya
+// Question Link: https://leetcode.com/problems/merge-sorted-array/
+
+
+class Solution {
+public:
+    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
+        int idx1 = m - 1; // Pointer for nums1.
+        int idx2 = n - 1; // Pointer for nums2.
+        int idx = m + n - 1; // Pointer for add number.
+
+        // Two-pointer method in valid indexes.
+        // Arrangement on nums1.
+        while(idx1 >= 0 && idx2 >= 0) {
+
+            // Add the larger number to end of the nums1.
+            if(nums1[idx1] >= nums2[idx2]) {
+                nums1[idx] = nums1[idx1];
+                idx1--; 
+            } else {
+                nums1[idx] = nums2[idx2];
+                idx2--;
+            }
+            idx--;
+        }
+        
+        // When numbers in nums1 are exhaused but nums2 still has some numbers,
+        // add remaining numbers from nums2 to beginning of nums1.
+        while(idx2 >= 0) {
+            nums1[idx] = nums2[idx2];
+            idx--;
+            idx2--;
+        }
```

Comment:
Approach and explanations are good but maybe you could add this code block after the last while loop to make your approach more understandable and explain why you did not consider the case which the nums1 exhausted and nums2 still has some numbers.
 
```c++
       /*
        unnecessary code block since nums1 already given sorted and if idx2 is exhausted
        it means that rest of the smaller values are in num1 array and they are already sorted.
        while(idx1 >= 0){
            nums1[idx] = nums1[idx1];
            idx--;
            idx1--;
        }
        */
```